src/main/java/org/owasp/esapi/reference/validation/HTMLValidationRule.java

```diff
@@ -97,8 +97,33 @@ public String sanitize( String context, String input ) {
 		return safe;
 	}
 
+    /**
+     * This admittedly is an UGLY hack to ensure that issue 509 and it's corresponding
+     * fix in PR #510 does not break existing developer's existing code. Full
+     * details are described in GitHub issue 521.
+     */
+    private boolean ignoreIssue509Fix() {
+        boolean ignore = true;
+        try {
+            // DISCUSS:
+            // Hindsight: maybe we should have getBooleanProp(), getStringProp(),
+            // getIntProp() methods that take a default arg as well?
+            // At least for ESAPI 3.x.
+            ignore = ESAPI.securityConfiguration().getBooleanProp(
+                            org.owasp.esapi.reference.DefaultSecurityConfiguration.VALIDATOR_IGNORE509 );
+        } catch( ConfigurationException cex ) {
+            // OPEN ISSUE: Should we log this? I think so. Convince me otherwise. But maybe
+            //             we should only log it once or every Nth time??
+            LOGGER.warning(Logger.EVENT_FAILURE, "ESAPI property " + 
+                           org.owasp.esapi.reference.DefaultSecurityConfiguration.VALIDATOR_IGNORE509 +
+                           " either unset or not set to a boolean value. Assuming 'true'.", cex);
+        }
+
+        return ignore;
+    }
+
 	private String invokeAntiSamy( String context, String input ) throws ValidationException {
-		// CHECKME should this allow empty Strings? "   " us IsBlank instead?
+		// CHECKME should this allow empty Strings? "   " use IsBlank instead?
```

Comment:
Agreed.  IsBlank.

Reply:
This was just a fixing a typo in the spelling. I'm not against changing this necessarily, but at this point, that could go beyond the original intent of this issue of just restoring the behavior to what it was before GitHub issue #509 was addressed in the PR that broke our JUnit tests.

If we want this changed, I think that someone ought to create a separate GitHub issue and do it under a separate PR. That way if we need to undo it for some reason (like, it turns out that it could break code again--and I doubt we have a test case for this), then it will make it easier to undo. Does that work for the two of you?

Also, BTW, there are CHECKME, DISCUSS, OPEN ITEM, and TODO comments littered throughout ESAPI that I have wanted someone to review and provide feedback on, but I've never previously had any takers. I probably added the majority of them. If either of you are feeling ambitious, don't stop at this one. :)